External/FEXCore/Source/Interface/HLE/Thunks/Thunks.cpp

```diff
@@ -29,8 +29,13 @@ tags: glue|thunks
 #include <stdint.h>
 #include <utility>
 
-#ifdef ENABLE_JEMALLOC
-#include "jemalloc/jemalloc.h"
+#ifdef ENABLE_JEMALLOC_GLIBC
+extern "C" {
+// jemalloc defines nothrow on its internal C function signatures.
+#define JEMALLOC_NOTHROW __attribute__((nothrow))
+// Forward declare jemalloc functions because we can't include the headers from the glibc jemalloc project.
```

Comment:
This comes down to cmake not being able to do per-file different header include directories.
Two independent jemalloc prefixes have the same include path `include/jemalloc/` and `pregen/include/jemalloc/`, we can't rename the paths otherwise the headers will fail to find some includes.
Since we are using two independent jemalloc implementations these include paths would intersect and means depending on search order we will get the `je_` prefixed function definitions or the `glibc_je_` prefixed function definitions depending.

Reply:
```suggestion
// Forward declare jemalloc functions because we can't include the headers from the glibc jemalloc project.
// This is because we can't simultaneously set up include paths for both of our internal jemalloc modules.
```

Reply:
Updated comment